Allows router to be aliased

markhuot opened the pull request:
This allows the `yii serve` command to accept an aliased router (as the document root is). An example of this could be routing to your `@app/web/index.php` file.

One use case for this is that, by default, `yii serve` will not route `.json` files through your router by default. However, if you use `yii serve --router="@app/web/index.php"` it will force all requests through the router, which will send `.json` requests through your application.

This is not a BC break because it does not change the default functionality (although I would argue we should make `$this->router = "@app/web/index.php";`). Instead it just supports that functionality for people who need it.

| Q             | A
| ------------- | ---
| Is bugfix?    | ✔️
| New feature?  | ❌
| Breaks BC?    | ❌
| Tests pass?   | ✔️
| Fixed issues  | n/a

markhuot committed:
Allows router to be aliased

Commit:
Merge branch 'master' into patch-1

samdark commented:
Code is alright.

1. Need a line for CHANGELOG
2. Need the info about this enhancement to be added at least to command description

Arhell committed:
fix intro-upgrade-from-v1.md link

rob006 commented on framework/console/controllers/ServeController.php:
```diff
@@ -52,6 +52,7 @@ class ServeController extends Controller
     public function actionIndex($address = 'localhost')
     {
         $documentRoot = Yii::getAlias($this->docroot);
+        $router = Yii::getAlias($this->router);
```
Shouldn't we check if `$this->router !== null` before using `Yii::getAlias()`?

markhuot replied:
Wouldn't the same logic apply for `$this->docroot`? I went for consistency. Should we check for both or just the router?

rob006 replied:
`docroot` has default value, which is already a string, and from console you can overwrite it only by another string. `router` is `null` by default, so we should expect that it will stay as `null`. So IMO this check is needed only for `router`.

winternet-studio committed:
Fixed typo

Arhell committed:
fix intro-upgrade-from-v1.md link (id)

RomanBush committed:
Update security-authorization.md (#18667)

change deprecated function to actual

datlechin committed:
Translate start-prerequisites.md into Vietnamese

Commit:
Replacing deprecated `::className()` with `::class` (#18671)

rhertogh committed:
Fix #18669: Changed visibility of `yii\web\User::checkRedirectAcceptable()` to `public`

Arhell committed:
fix intro-yii.md (fr) link

aidanbek committed:
Fix #18678: Fix `yii\caching\DbCache` to use configured cache table name instead of the default one in case of MSSQL varbinary column type detection

Commit:
Add more tests for helpers (#18675)

Aresenka committed:
Fix #18637: Mentioned that relation can not be named "relation"

Arhell committed:
Fix typo

rhertogh committed:
Fix #18676: Added method `yii\helpers\BaseFileHelper::changeOwnership()` and properties `newFileMode`/`newFileOwnership` in `yii\console\controllers\BaseMigrateController`

Co-authored-by: Bizley <pawel@positive.codes>

Arhell committed:
Fix id intro-yii.md (id) link (#18688)

Commit:
Scrutinizer fixes (#18685)

* https://scrutinizer-ci.com/g/yiisoft/yii2/issues/master/files/framework/filters/AccessRule.php?selectedLabels%5B0%5D=2&selectedSeverities%5B0%5D=10&selectedSeverities%5B1%5D=5&orderField=path&order=asc&honorSelectedPaths=0

* https://scrutinizer-ci.com/g/yiisoft/yii2/issues/master/files/framework/helpers/BaseFileHelper.php?selectedLabels%5B0%5D=2&selectedSeverities%5B0%5D=10&selectedSeverities%5B1%5D=5&orderField=path&order=asc&honorSelectedPaths=0

* https://scrutinizer-ci.com/g/yiisoft/yii2/issues/master/files/framework/base/Application.php?selectedLabels%5B0%5D=7&selectedSeverities%5B0%5D=10&selectedSeverities%5B1%5D=5&orderField=path&order=asc&honorSelectedPaths=0

* https://scrutinizer-ci.com/g/yiisoft/yii2/issues/master/files/framework/base/Module.php?selectedLabels%5B0%5D=7&selectedSeverities%5B0%5D=10&selectedSeverities%5B1%5D=5&orderField=path&order=asc&honorSelectedPaths=0

* https://scrutinizer-ci.com/g/yiisoft/yii2/issues/master/files/framework/web/MultipartFormDataParser.php?selectedLabels%5B0%5D=7&selectedSeverities%5B0%5D=10&selectedSeverities%5B1%5D=5&orderField=path&order=asc&honorSelectedPaths=0

* https://scrutinizer-ci.com/g/yiisoft/yii2/issues/master/files/framework/web/Session.php?selectedLabels%5B0%5D=7&selectedSeverities%5B0%5D=10&selectedSeverities%5B1%5D=5&orderField=path&order=asc&honorSelectedPaths=0

* https://scrutinizer-ci.com/g/yiisoft/yii2/issues/master/files/framework/base/Controller.php?selectedLabels%5B0%5D=6&selectedSeverities%5B0%5D=10&selectedSeverities%5B1%5D=5&orderField=path&order=asc&honorSelectedPaths=0

* https://scrutinizer-ci.com/g/yiisoft/yii2/issues/master/files/framework/console/controllers/MigrateController.php?selectedLabels%5B0%5D=6&selectedSeverities%5B0%5D=10&selectedSeverities%5B1%5D=5&orderField=path&order=asc&honorSelectedPaths=0

* Added vardocs, and small refactor

* https://scrutinizer-ci.com/g/yiisoft/yii2/issues/master/files/framework/db/conditions/InConditionBuilder.php?selectedLabels%5B0%5D=6&selectedSeverities%5B0%5D=10&selectedSeverities%5B1%5D=5&orderField=path&order=asc&honorSelectedPaths=0

* https://scrutinizer-ci.com/g/yiisoft/yii2/issues/master/files/framework/db/Connection.php?selectedLabels%5B0%5D=6&selectedSeverities%5B0%5D=10&selectedSeverities%5B1%5D=5&orderField=path&order=asc&honorSelectedPaths=0

* https://scrutinizer-ci.com/g/yiisoft/yii2/issues/master/files/framework/db/mssql/QueryBuilder.php?selectedLabels%5B0%5D=6&selectedSeverities%5B0%5D=10&selectedSeverities%5B1%5D=5&orderField=path&order=asc&honorSelectedPaths=0

* https://scrutinizer-ci.com/g/yiisoft/yii2/issues/master/files/framework/log/Logger.php?selectedLabels%5B0%5D=6&selectedSeverities%5B0%5D=10&selectedSeverities%5B1%5D=5&orderField=path&order=asc&honorSelectedPaths=0

* https://scrutinizer-ci.com/g/yiisoft/yii2/issues/master/files/framework/console/controllers/BaseMigrateController.php?selectedLabels%5B0%5D=18&selectedSeverities%5B0%5D=10&selectedSeverities%5B1%5D=5&orderField=path&order=asc&honorSelectedPaths=0

* Adding space between type

* https://scrutinizer-ci.com/g/yiisoft/yii2/issues/master/files/framework/console/controllers/CacheController.php?selectedLabels%5B0%5D=18&selectedSeverities%5B0%5D=10&selectedSeverities%5B1%5D=5&orderField=path&order=asc&honorSelectedPaths=0

* https://scrutinizer-ci.com/g/yiisoft/yii2/issues/master/files/framework/console/controllers/FixtureController.php?selectedLabels%5B0%5D=18&selectedSeverities%5B0%5D=10&selectedSeverities%5B1%5D=5&orderField=path&order=asc&honorSelectedPaths=0

* https://scrutinizer-ci.com/g/yiisoft/yii2/issues/master/files/framework/console/controllers/MigrateController.php?selectedLabels%5B0%5D=18&selectedSeverities%5B0%5D=10&selectedSeverities%5B1%5D=5&orderField=path&order=asc&honorSelectedPaths=0

* Update framework/console/controllers/CacheController.php

Co-authored-by: Alexander Makarov <sam@rmcreative.ru>

* Update framework/console/controllers/CacheController.php

Co-authored-by: Alexander Makarov <sam@rmcreative.ru>

* Update framework/console/controllers/FixtureController.php

Co-authored-by: Alexander Makarov <sam@rmcreative.ru>

* Update framework/console/controllers/FixtureController.php

Co-authored-by: Alexander Makarov <sam@rmcreative.ru>

* Update framework/console/controllers/FixtureController.php

Co-authored-by: Alexander Makarov <sam@rmcreative.ru>

* Update framework/console/controllers/FixtureController.php

Co-authored-by: Alexander Makarov <sam@rmcreative.ru>

* Update framework/console/controllers/FixtureController.php

Co-authored-by: Alexander Makarov <sam@rmcreative.ru>

* Update framework/console/controllers/FixtureController.php

Co-authored-by: Alexander Makarov <sam@rmcreative.ru>

Co-authored-by: Alexander Makarov <sam@rmcreative.ru>

Arhell committed:
Fix intro-yii.md (pt-BR) link (#18692)

Arhell committed:
fix intro-yii.md (uz) link

Commit:
Fix some code smells (#18696)

rhertogh committed:
Fix #18695: Added `yii\web\Cookie::SAME_SITE_NONE` constant

samdark committed:
Fix console error handler exception handling condition. Related to #18696

See https://github.com/yiisoft/yii2/pull/18696#discussion_r645877103

Arhell committed:
fix intro-yii.md (zh-CN) link

Arhell committed:
fix intro-yii.md (vi) link

Arhell committed:
fix IIS Web site link

dicrtarasov committed:
fix Exception\Throwable order

dicrtarasov committed:
fix QueryInterface::count return type

Arhell committed:
fix info block

RodrigoDornelles committed:
use getTargetClass in function checkTargetAttributeExistence

This interesting change, to be able to change getTargetClass when necessary when extending this class.

Arhell committed:
update info block in start-hello.md

Arhell committed:
update info block all .md (uz)

winternet-studio committed:
Added example with hyphen in action ID

Also added link to guide on actions - don't know if you have a better way than providing a static URL... And maybe it should be moved to the class description?

marcovtwout committed:
Allow jQuery 3.6 to be installed
See https://blog.jquery.com/2021/03/02/jquery-3-6-0-released/

Arhell committed:
single style to structure block (start-workflow.md)

Commit:
Update security-passwords.md to more correct word

Fundamentally, hashes cannot be `reversed`, rather they are cracked.

Commit:
Update docs/guide/security-passwords.md

Co-authored-by: Bizley <pawel@positive.codes>

Arhell committed:
fix books link & update code block

rhertogh committed:
Json pretty print (#18726)

* Added pretty print support to Json helper

* Updated documentation for `yii\helpers\Json::$prettyPrint`

* Fixed line ending in \yiiunit\framework\helpers\JsonTest::testPrettyPrint for windows

* Added helper-json.md link in docs/guide/README.md

* Capitalized JSON in descriptions

Co-authored-by: Alexander Makarov <sam@rmcreative.ru>

Co-authored-by: Alexander Makarov <sam@rmcreative.ru>
Co-authored-by: Bizley <pawel@positive.codes>

Arhell committed:
update IRC chat (#18736)

olegbaturin committed:
Fix #18648: Fix `yii\web\Request` to properly handle HTTP Basic Auth headers

vjik committed:
Adopt readme logo to dark theme (#18739)

egorrishe committed:
Fix #18737: Fix SchemaTest::testColumnSchema() for MySql <5.7

Commit:
Fix #18712: Added `$scheme` option to `yii\i18n\Formatter::asUrl()`

CoolGoose committed:
Only log the $_SERVER information under YII_DEBUG (#18707)

* Only log the $_SERVER information under YII_DEBUG

Since $_SERVER can contain sensitive information that could leak in CI pipelines ideally it should happen under the YII_DEBUG flag only

* Added upgrade / changelog

* Corrected the upgrade information

* Update framework/UPGRADE.md

Co-authored-by: Alexander Makarov <sam@rmcreative.ru>

Co-authored-by: Bizley <pawel@positive.codes>
Co-authored-by: Alexander Makarov <sam@rmcreative.ru>

Arhell committed:
update forum link (start-looking-ahead.md) (#18744)

Commit:
Updated asUrl method to not introduce BC break (#18742)

brandonkelly committed:
Fix #18734: Added `yii\validators\EmailValidator::$enableLocalIDN`

mrblc committed:
Fixed tipfellery and inconsistencies regardin file translation for file validation messages in Croatian

egorrishe committed:
don't convert int to string if db type of column is numeric (#18741)

* write adequate test for issue #14663

* don't convert int to string if db type of column is numeric (#14663)

* fix bigint schema test for MySql >8.0.17 (#14663)

* Update CHANGELOG.md

* Update CHANGELOG.md

* update phpdoc [ci skip] (#14663)

* refactoring test case to make it clearer [ci skip] (#14663)

* check `int unsigned` in `QueryBuilderTest::testInsertInteger()` (#14663)

* Update Upgrade.md (#14663)

* fix `int unsigned` schema test for MySql >8.0.17 (#14663)

* fix `int unsigned` schema test for MySql <5.7 (#14663)

Co-authored-by: Alexander Makarov <sam@rmcreative.ru>
Co-authored-by: Bizley <pawel@positive.codes>

vjik committed:
Fix #18749: Fix `yii\web\ErrorHandler::encodeHtml()` to support strings with invalid UTF symbols

stevekr committed:
Fix $value parameter types in yii\i18n\Formatter::normalizeDatetimeValue()

Arhell committed:
update yii forum & chat link (#18751)

Arhell committed:
update structure block (#18760)

Arhell committed:
update composer.json links

Arhell committed:
update http to https website links

kayw-geek committed:
Change the link protocol to HTTPS

Arhell committed:
update twig link (#18774)

Co-authored-by: Bizley <pawel@positive.codes>

Arhell committed:
update fontawesome link (#18780)

SOHELAHMED7 committed:
Make docs gender agnostics (#18781)

YuraPlinto committed:
Fix typo in Russian translation of the guide

Commit:
PL Guide updates (#18787)

bizley commented:
@markhuot can we get the changes requested by rob and samdark, please?

markhuot committed:
documentation update

markhuot committed:
documentation update

markhuot committed:
Merge remote-tracking branch 'origin/master' into patch-1

markhuot committed:
fixing bad commit
